File: app/src/main/java/agersant/polaris/api/remote/IRemoteAPI.java
package agersant.polaris.api.remote;

import android.net.Uri;

import java.io.IOException;

import agersant.polaris.api.IPolarisAPI;
import agersant.polaris.api.ItemsCallback;
import agersant.polaris.api.ThumbnailSize;
import okhttp3.ResponseBody;

public interface IRemoteAPI extends IPolarisAPI {

    void getRandomAlbums(ItemsCallback handlers);

    void getRecentAlbums(ItemsCallback handlers);

    void setLastFMNowPlaying(String path);

    void scrobbleOnLastFM(String path);

    Uri getAudioUri(String path);

    Uri getThumbnailUri(String path, ThumbnailSize size);

    ResponseBody getAudio(String path) throws IOException;

    ResponseBody getThumbnail(String path, ThumbnailSize size) throws IOException;
}

File: app/src/main/java/agersant/polaris/api/remote/IRemoteAPI.kt
package agersant.polaris.api.remote

import agersant.polaris.Directory
import agersant.polaris.api.IPolarisAPI
import agersant.polaris.api.ThumbnailSize
import android.graphics.Bitmap
import android.net.Uri

interface IRemoteAPI : IPolarisAPI {

    suspend fun getRandomAlbums(): List<Directory>?

    suspend fun getRecentAlbums(): List<Directory>?

    suspend fun setLastFmNowPlaying(path: String): Boolean

    suspend fun scrobbleOnLastFm(path: String): Boolean

    suspend fun getAudioUri(path: String): Uri?

    suspend fun getThumbnail(path: String, size: ThumbnailSize): Bitmap?
}
